packages/formstr-app/src/containers/CreateFormNew/components/QuestionCard/index.tsx

```diff
@@ -77,27 +78,55 @@ const QuestionCard: React.FC<QuestionCardProps> = ({
           firstQuestion={firstQuestion}
           lastQuestion={lastQuestion}
         />
-        <div className="question-text">
-          <QuestionTextStyle>
+        <div className="question-text" style={{justifyContent: "space-between", display:"flex"}} >
+          <QuestionTextStyle style={{ width: "100%" }}>
             <label>
               <Input.TextArea
+                key={question[1]} 
                 className="question-input"
                 onChange={handleTextChange}
-                defaultValue={question[3] || "Click to edit"}
+                value={question[3] || "Click to edit"}
                 placeholder="Enter a Question"
                 autoSize
               />
             </label>
           </QuestionTextStyle>
+          <UploadImage 
+            onImageUpload={(markdownUrl) => {
+              const name = markdownUrl.match(/\[(.*?)\]/)?.[1] || '';
+              const url = markdownUrl.match(/\((.*?)\)/)?.[1] || '';
+              const imageMarkdown = `![${name}](${url})`;
+            
+              const currentDisplay = question[3] || '';
+              const newDisplay = currentDisplay 
+                ? `${currentDisplay}\n\n${imageMarkdown}`
+                : imageMarkdown;
+            
+              const field: Field = [
+                question[0],
+                question[1],
+                question[2],
+                newDisplay,
+                question[4],
+                JSON.stringify({
+                  ...answerSettings,
+                  imageUrl: imageMarkdown,
```

Comment:
we don't need this anymore? since we're already adding the image markdown in `newDisplay` on `question[3]` ?

Reply:
yes, we don't need this!